examples/GLUE/src/data.py

```diff
@@ -0,0 +1,128 @@
+        df = pd.DataFrame(self.datasets[split])
+        print(df.head(2))
+        count = df.groupby('label')['label'].count()
+        summary = pd.DataFrame({'count':count, 'ratio': count/len(df)})
+        print(summary)
+        return df
+    
+    def analyze_texts(self, texts):
+        s = pd.Series( texts)
+        s = s.apply(lambda cell : cell.split())
+        print(s.apply(len).describe(percentiles = [0.5,0.95,0.99,0.999]))
+
+class SentenceDataAnalyzer(GlueDataAnalyzer):
+    def __init__(self, glue_task, sentence_key = 'sentence'):
+        super().__init__(glue_task)
+        self.sentence_key = sentence_key
+    def analyze_split(self, split = 'train'):
+        df = super().analyze_split(split)
+        self.analyze_texts(df[self.sentence_key])
+
+class SPDataAnalyzer(GlueDataAnalyzer):
+    def __init__(self, glue_task, s1_key = 'question1', s2_key = 'question2'):
+        super().__init__(glue_task)
+        self.s1_key = s1_key
+        self.s2_key = s2_key
+    def analyze_split(self, split = 'train'):
+        df = super().analyze_split(split)
+        self.analyze_texts(df[self.s1_key])
+        self.analyze_texts(df[self.s2_key])
+
+
+class SPRegressionDataAnalyzer(SPDataAnalyzer):
+    def __init__(self, glue_task, s1_key = 'question1', s2_key = 'question2'):
+        super().__init__(glue_task)
+        self.s1_key = s1_key
+        self.s2_key = s2_key
+    def analyze_split(self, split = 'train'):
+        print(f'\n{split} data label distribution')
+        df = pd.DataFrame(self.datasets[split])
+        print(df.head(2))
+        # print(df.label.describe())
+        self.analyze_texts(df[self.s1_key])
+        self.analyze_texts(df[self.s2_key])
+
+def analyzer_factory(glue_task):
+    factory = {
+        'default':GlueDataAnalyzer(glue_task),
+        'qqp':SPDataAnalyzer('qqp'),
+        'rte':SPDataAnalyzer('rte', 'sentence1', 'sentence2'),
+        'mnli':SPDataAnalyzer('mnli', 'premise', 'hypothesis'),
+        'qnli':SPDataAnalyzer('qnli', 'question', 'sentence'),
+        'sst2':SentenceDataAnalyzer('sst2'),
+        'stsb':SPRegressionDataAnalyzer('stsb', 'sentence1','sentence2'),
+        'wnli':SPDataAnalyzer('wnli', 'sentence1', 'sentence2'),
+        'mrpc':SPDataAnalyzer('mrpc', 'sentence1', 'sentence2'),
+    }
+    glue_task = glue_task if glue_task in factory else 'default'
+    return factory[glue_task]
+
+class GlueData(DataInterface):
+    def setup_datasets(self, glue_task = 'cola'):
+        self.glue_task = glue_task
+        datasets = load_dataset("glue",glue_task, cache_dir = cache_dir)
+        self.train_ds = datasets['train']
+        self.val_ds = datasets['validation']
+        self.test_ds = datasets['test']
+    
+    def get_train_dataset(self):
+        return self.train_ds
+    
+    def get_val_dataset(self):
+        return self.val_ds 
+
+    def get_test_dataset(self):
+        return self.test_ds
+
+class MNLIData(GlueData):
```

Comment:
It seems like a separate class and factory for MNLI can be avoided by adding a single if condition in the datainterface for self.val_ds and self.test_ds